### ISSUE TITLE
TypeError: Argument 1 passed to Drush\Drupal\Commands\core\LocaleCommands after upgrade to Drush 9.5.2

### ISSUE THREAD

**Issue by dhendriks** [drush-ops/drush#3753]
After upgrading from Drush 9.4.0 to 9.5.2, I now get the following error when running drush with a drupal instance set via --root (there is no error if i run drush without --root):

```
 [error]    ::__construct() must implement interface Drupal\Core\Language\LanguageManagerInterface, instance of Drupal\Core\Extension\ModuleHandler given, called in <DRUPAL_PATH>/public_html/core/lib/Drupal/Component/DependencyInjection/Container.php on line 266 in Drush\Drupal\Commands\core\LocaleCommands->__construct() (line 59 of <DRUSH_PATH>/vendor/drush/drush/src/Drupal/Commands/core/LocaleCommands.php)
#0 <DRUPAL_PATH>/public_html/core/lib/Drupal/Component/DependencyInjection/Container.php(266): Drush\Drupal\Commands\core\LocaleCommands->__construct(Object(Drupal\Core\Extension\ModuleHandler), Object(Drupal\Core\State\State))
#1 <DRUPAL_PATH>/public_html/core/lib/Drupal/Component/DependencyInjection/Container.php(171): Drupal\Component\DependencyInjection\Container->createService(Array, 'locale.commands')
#2 <DRUPAL_PATH>/public_html/core/lib/Drupal/Component/DependencyInjection/Container.php(480): Drupal\Component\DependencyInjection\Container->get('locale.commands', 1)
#3 <DRUPAL_PATH>/public_html/core/lib/Drupal/Component/DependencyInjection/Container.php(319): Drupal\Component\DependencyInjection\Container->resolveServicesAndParameters(Array)
#4 <DRUPAL_PATH>/public_html/core/lib/Drupal/Component/DependencyInjection/Container.php(171): Drupal\Component\DependencyInjection\Container->createService(Array, 'drush.command.s...')
#5 <DRUSH_PATH>/vendor/drush/drush/src/Boot/DrupalBoot8.php(230): Drupal\Component\DependencyInjection\Container->get('drush.command.s...')
#6 <DRUSH_PATH>/vendor/drush/drush/src/Boot/BootstrapManager.php(295): Drush\Boot\DrupalBoot8->bootstrapDrupalFull(NULL)
#7 <DRUSH_PATH>/vendor/drush/drush/src/Boot/BootstrapManager.php(504): Drush\Boot\BootstrapManager->doBootstrap(5, 6, NULL)
#8 <DRUSH_PATH>/vendor/drush/drush/src/Boot/BootstrapManager.php(438): Drush\Boot\BootstrapManager->bootstrapMax(6)
#9 <DRUSH_PATH>/vendor/drush/drush/src/Boot/BootstrapManager.php(407): Drush\Boot\BootstrapManager->bootstrapToPhaseIndex(-2, Object(Consolidation\AnnotatedCommand\AnnotationData))
#10 <DRUSH_PATH>/vendor/drush/drush/src/Boot/BootstrapHook.php(28): Drush\Boot\BootstrapManager->bootstrapToPhase('max', Object(Consolidation\AnnotatedCommand\AnnotationData))
#11 <DRUSH_PATH>/vendor/consolidation/annotated-command/src/Hooks/Dispatchers/InitializeHookDispatcher.php(34): Drush\Boot\BootstrapHook->initialize(Object(Drush\Symfony\DrushArgvInput), Object(Consolidation\AnnotatedCommand\AnnotationData))
#12 <DRUSH_PATH>/vendor/consolidation/annotated-command/src/Hooks/Dispatchers/InitializeHookDispatcher.php(27): Consolidation\AnnotatedCommand\Hooks\Dispatchers\InitializeHookDispatcher->callInitializeHook(Object(Drush\Boot\BootstrapHook), Object(Drush\Symfony\DrushArgvInput), Object(Consolidation\AnnotatedCommand\AnnotationData))
#13 <DRUSH_PATH>/vendor/consolidation/annotated-command/src/CommandProcessor.php(117): Consolidation\AnnotatedCommand\Hooks\Dispatchers\InitializeHookDispatcher->initialize(Object(Drush\Symfony\DrushArgvInput), Object(Consolidation\AnnotatedCommand\AnnotationData))
#14 <DRUSH_PATH>/vendor/consolidation/annotated-command/src/AnnotatedCommand.php(391): Consolidation\AnnotatedCommand\CommandProcessor->initializeHook(Object(Drush\Symfony\DrushArgvInput), Array, Object(Consolidation\AnnotatedCommand\AnnotationData))
#15 <DRUPAL_PATH>/vendor/symfony/console/Command/Command.php(221): Consolidation\AnnotatedCommand\AnnotatedCommand->initialize(Object(Drush\Symfony\DrushArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#16 <DRUPAL_PATH>/vendor/symfony/console/Application.php(964): Symfony\Component\Console\Command\Command->run(Object(Drush\Symfony\DrushArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#17 <DRUPAL_PATH>/vendor/symfony/console/Application.php(248): Symfony\Component\Console\Application->doRunCommand(Object(Consolidation\AnnotatedCommand\AnnotatedCommand), Object(Drush\Symfony\DrushArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#18 <DRUPAL_PATH>/vendor/symfony/console/Application.php(148): Symfony\Component\Console\Application->doRun(Object(Drush\Symfony\DrushArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#19 <DRUSH_PATH>/vendor/drush/drush/src/Runtime/Runtime.php(112): Symfony\Component\Console\Application->run(Object(Drush\Symfony\DrushArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#20 <DRUSH_PATH>/vendor/drush/drush/src/Runtime/Runtime.php(41): Drush\Runtime\Runtime->doRun(Array)
#21 <DRUSH_PATH>/vendor/drush/drush/drush.php(66): Drush\Runtime\Runtime->run(Array)
#22 <DRUSH_PATH>/vendor/drush/drush/drush(4): require('<HOME_PATH>...')
#23 {main}.
```

I'm using Drupal core 8.6.1

**Comment by rroblik** [comment]
> 
> 
> I have core 8.6.2, but I have a same error

+1

**Comment by itcrowd72** [comment]
Hm... **drush cr** fix my problem

**Comment by dhendriks** [comment]
> Hm... **drush cr** fix my problem

Thanks for letting us now. A bit weird though, as for me **drush --help** already failed, assuming that *--root** is used.

**Comment by weitzman** [comment]
Are there folks who have run `drush cr` after updating Drush and are still getting this error? `drush cr` is required after updating Drush as we store command info in the Drupal container, and that info may have gone stale after code change.

**Comment by Renrhaf** [comment]
This error disappeared after a proper composer update + cache clearing for me. Thanks.

**Comment by gitressa** [comment]
If you run `composer update --no-dev` twice, the second time it downgrades drush to 9.4, if there is no constraint on Drush, which I think might also have been the reason I got an error message.
```
Package operations: 1 install, 2 updates, 0 removals
  - Installing symfony/polyfill-php72 (v1.10.0): Downloading (100%)         
  - Updating symfony/var-dumper (v3.4.20 => v4.2.1): Downloading (100%)         
  - Downgrading drush/drush (9.5.2 => 9.4.0): Downloading (100%)         
```
For more, see https://github.com/drupal-composer/drupal-project/issues/444

@webflo's explanation helped me understand:
> Its not a composer problem. This happens because the drush has less strict dependencies in version 9.4.0 than in 9.5.2. If you want to build a site without the dev packages you should run composer install --no-dev. Or use a stricter version constraint for drush. This issue will be resolved once drush 9.6.0 has been released, because it supports Symfony 3 and 4.
